crates/sui-types/src/object.rs

```diff
@@ -72,6 +69,7 @@ impl MoveObject {
     pub unsafe fn new_from_execution(
         type_: StructTag,
         has_public_transfer: bool,
+        version: SequenceNumber,
```

Comment:
This is probably the biggest change for MoveObject. Lots of downstream changes as a result of this